handlers.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+
+type APIServer struct {
+	listenAddress string
+	database Database
+}
+
+func NewAPIServer(listenAddress string, database Database) *APIServer {
+	return &APIServer{
+		listenAddress: listenAddress,
+		database: database,
+	}
+}
+
+func (s *APIServer) Run() {
+	r := mux.NewRouter()
+
+	r.HandleFunc("/item", makeHTTPHandleFunc(s.handleItem))
+	r.HandleFunc("/item/{id}", makeHTTPHandleFunc(s.listItemHandler))
+
+	// endpoints, handler functions and HTTP methods
+	// r.HandleFunc("/items", listItemsHandler).Methods("GET")
+	// r.HandleFunc("/items", createItemHandler).Methods("POST")
+	// r.HandleFunc("/items/{id}", updateItemHandler).Methods("PUT")
+	// r.HandleFunc("/items/{id}", deleteItemHandler).Methods("DELETE")
+	// r.HandleFunc("/items/{id}", listItemHandler).Methods("GET")
+
+	http.ListenAndServe(s.listenAddress, r)
+}
+
+func (s *APIServer) handleItem(w http.ResponseWriter, r *http.Request) error {
+	if r.Method == "GET" {
+		return s.listItemsHandler(w, r)
+	}
+	if r.Method == "POST" {
+		return s.createItemHandler(w, r)
+	}
+	if r.Method == "DELETE" {
+		return s.deleteItemHandler(w, r)
+	}
+	if r.Method == "PUT" {
+		return s.updateItemHandler(w, r)
+	}
+	return nil
+}
+
+func (s *APIServer) listItemsHandler(w http.ResponseWriter, r *http.Request) error {
+	items, err := s.database.ListItems()
+	if err != nil {
+		return err
```

Comment:
como é um http handler precisa ter um write com bad request , not found ou internal server error para tratamento desse erro